sdk/eventgrid/azure-messaging-eventgrid-namespaces/customization/src/main/java/EventGridCustomization.java

```diff
@@ -63,5 +38,19 @@ public void customizeEventGridClientImplImports(LibraryCustomization customizati
         customization.getRawEditor().removeFile("src/main/java/com/azure/messaging/eventgrid/namespaces/models/PublishResult.java");
         customization.getRawEditor().removeFile("src/main/java/com/azure/messaging/eventgrid/namespaces/models/CloudEvent.java");
 
+        // add preview ServiceVersion
+        PackageCustomization packageCustomization = customization.getPackage("com.azure.messaging.eventgrid.namespaces");
+        ClassCustomization classCustomization = packageCustomization.getClass("EventGridServiceVersion");
+        classCustomization.customizeAst(compilationUnit -> {
+            EnumDeclaration clazz = compilationUnit.getEnumByName("EventGridServiceVersion").get();
+            clazz.getEntries().add(0, new EnumConstantDeclaration("/**\n" +
+                " * Enum value 2023-10-01-preview.\n" +
+                " */\n" +
+                "V2023_10_01_PREVIEW(\"2023-10-01-preview\")"));
+            clazz.getEntries().add(0, new EnumConstantDeclaration("/**\n" +
+                " * Enum value 2023-06-01-preview.\n" +
+                " */\n" +
+                "V2023_06_01_PREVIEW(\"2023-06-01-preview\")"));
+        });
```

Comment:
I've use the customization to add back the preview ServiceVersion.

However, I think they should not be included in GA SDK release.